pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -191,7 +200,9 @@ protected Message internalReceive(int timeout, TimeUnit unit) throws PulsarClien
         Message message;
         try {
             lock.writeLock().lock();
-            message = incomingMessages.poll(0, TimeUnit.SECONDS);
+            message = (incomingMessages.size() > 0) ?
```

Comment:
I realize this is not good idea, but I faced the problem that messages were on `incomingMessages` of any consumer level but the `incomingMessages` at partitioned level remained empty, so the consumer goes to 'pendingReceives' without handling messages in `incomingMessages` at consumer impl level.